Mathlib/Order/RelSeries.lean

```diff
@@ -0,0 +1,261 @@
+
+/-- Relation series of `r` and nonempty list of `α` satisfying `r`-chain condition bijectively
+corresponds to each other.-/
+protected def Equiv : RelSeries r ≃ {x : List α | x ≠ ∅ ∧ x.Chain' r} where
+  toFun := fun x => ⟨_, x.toList_ne_empty, x.toList_chain'⟩
+  invFun := fun x => fromListChain' _ x.2.1 x.2.2
+  left_inv := fun x => ext (by dsimp; rw [List.length_ofFn, Nat.pred_succ]) <| by
+    ext f
+    simp only [List.empty_eq, ne_eq, Set.coe_setOf, Set.mem_setOf_eq, fromListChain'_toFun,
+      Function.comp_apply, List.get_ofFn, fromListChain'_length]
+    rfl
+  right_inv := by
+    intro x
+    refine Subtype.ext (List.ext_get ?_ <| fun n hn1 hn2 => ?_)
+    · dsimp
+      rw [List.length_ofFn, fromListChain'_length, ←Nat.succ_eq_add_one, Nat.succ_pred_eq_of_pos]
+      rw [List.length_pos]
+      exact x.2.1
+    · rw [List.get_ofFn, fromListChain'_toFun, Function.comp_apply]
+      congr
+
+-- TODO : build a similar bijection between `RelSeries α` and `Quiver.Path`
+
+end RelSeries
+
+namespace Rel
+
+/-- a relation `r` is said to be finite dimensional iff there is a relation series of `r` with the
+  maximum length. -/
+class FiniteDimensional where
+  /-- the longest relation series-/
+  longestRelSeries : RelSeries r
+  /-- the longest relation series is longer than any other series -/
+  is_longest : ∀ (y : RelSeries r), y.length ≤ longestRelSeries.length
+
+/-- the longest relational series when a relation is finite dimensional -/
+def longestRelSeries [r.FiniteDimensional] : RelSeries r := FiniteDimensional.longestRelSeries
+
+lemma longestRelSeries_is_longest [r.FiniteDimensional] (x : RelSeries r) :
+  x.length ≤ r.longestRelSeries.length := FiniteDimensional.is_longest _
+
+/-- a relation `r` is said to be infinite dimensional iff there exists relation series of arbitrary
+  length. -/
+class InfiniteDimensional where
+  /-- there should a relation series with length at least `n` for all natural number `n` -/
+  longRelSeries : ℕ → RelSeries r
+  /-- there should a relation series with length at least `n` for all natural number `n` -/
+  longRelSeries_length : ∀ (n : ℕ), n ≤ (longRelSeries n).length
+
+/-- a relation series with length at least `n` if the relation is infinite dimensional -/
+def relSeriesOfInfiniteDimensional [InfiniteDimensional r] (n : ℕ) : RelSeries r :=
+  (InfiniteDimensional.longRelSeries n)
+
+lemma le_relSeriesOfInfiniteDimensional_length [InfiniteDimensional r] (n : ℕ) :
+  n ≤ (r.relSeriesOfInfiniteDimensional n).length := InfiniteDimensional.longRelSeries_length n
+
+/-- if a relation on `α` is infinite dimensional, then `α` is inhabited -/
+def inhabited_of_infiniteDimensional [r.InfiniteDimensional] : Inhabited α :=
+  ⟨(r.relSeriesOfInfiniteDimensional 0) 0⟩
+
+end Rel
+
+/-- A type is finite dimensional if its `LTSeries` has bounded length. -/
+class FiniteDimensionalType (γ : Type _) [Preorder γ] extends
+  Rel.FiniteDimensional ((. < .) : γ → γ → Prop)
```

Comment:
I think
```suggestion
abbrev FiniteDimensionalType (γ : Type _) [Preorder γ] :=
  Rel.FiniteDimensional ((. < .) : γ → γ → Prop)
```
would be reasonable. (Same below)

Reply:
changed